pkg/occlient/occlient.go

```diff
@@ -1290,22 +1302,42 @@ func (c *Client) PatchCurrentDC(name string, dc appsv1.DeploymentConfig, prePatc
 	}
 
 	// Replace the current spec with the new one
-	currentDC.Spec = dc.Spec
+	modifiedDC.Spec = dc.Spec
 
 	// Replace the old annotations with the new ones too
 	// the reason we do this is because Kubernetes handles metadata such as resourceVersion
 	// that should not be overridden.
-	currentDC.ObjectMeta.Annotations = dc.ObjectMeta.Annotations
-	currentDC.ObjectMeta.Labels = dc.ObjectMeta.Labels
+	modifiedDC.ObjectMeta.Annotations = dc.ObjectMeta.Annotations
+	modifiedDC.ObjectMeta.Labels = dc.ObjectMeta.Labels
 
 	// Update the current one that's deployed with the new Spec.
 	// despite the "patch" function name, we use update since `.Patch` requires
 	// use to define each and every object we must change. Updating makes it easier.
-	_, err := c.appsClient.DeploymentConfigs(c.Namespace).Update(currentDC)
+	updatedDc, err := c.appsClient.DeploymentConfigs(c.Namespace).Update(&modifiedDC)
 	if err != nil {
 		return errors.Wrapf(err, "unable to update DeploymentConfig %s", name)
 	}
 
+	// till we find a better solution for git
+	// it is better, not to follow the logs as the build will anyways trigger a new deployment later
+	if !waitForDc {
+		return nil
+	}
+
+	// we check after the update that the template in the earlier and the new dc are same or not
+	// if they are same, we don't wait as new deployment won't run and we will wait till timeout
+	// inspired from https://github.com/openshift/origin/blob/bb1b9b5223dd37e63790d99095eec04bfd52b848/pkg/apps/controller/deploymentconfig/deploymentconfig_controller.go#L609
+	if reflect.DeepEqual(updatedDc.Spec.Template, currentDC.Spec.Template) {
+		return nil
+	} else {
+		currentDCBytes, err := json.Marshal(currentDC.Spec.Template)
+		updatedDCBytes, err := json.Marshal(updatedDc.Spec.Template)
+		if err != nil {
+			return errors.Wrapf(err, "unable to unmarshal dc")
+		}
+		glog.V(4).Infof("updatedDc Spec.Template: %v, not matching currentDc Spec.Template: %v", string(updatedDCBytes), string(currentDCBytes))
```

Comment:
it would be good to add a  little bit more info to that message, like what it means if this is not matching. If I would read just this message in logs and didn't know how this is working I would think that this means that something went wrong